### PR TITLE
Effects revamp

### LINKED ISSUE #54: Current way of saving score in json doesnt allow multiple people with the same scores
The name will be replaced instead of adding a new entry

### LINKED ISSUE #56: Shrink and enlarge should only grow in height and not in width

### PR DESCRIPTION
**Pull request by ivantjh**
No implementation of new effects but the basic effects are more consistent now. Use queues to manage effects. 

Fixes #54
Fixes #56

### CODE REVIEW

File: src/effects.cpp [hunk 1]
```diff
@@ -10,15 +14,46 @@ void Effects::update(float frameTime) {
...
 // Add new effect
 void Effects::addEffect(effectNS::EFFECT_TYPE effectType, float duration) {
-	currentEffects.insert({ effectType, duration });
+	// If effect is already running, do not re-initialize effects
+	// For accurate timings, initialize effect, then start timers
+	bool effectExists = false;
+
+	for (effectDurationPair curr : currentEffects) {
+		if (curr.first == effectType)
+			effectExists = true;
```
Comment (chesterhow) [inline src/effects.cpp]: `effectExists = (curr.first == effectType);`

File: src/entity.cpp [hunk 2]
```diff
@@ -444,6 +444,8 @@ void Entity::addEffect(effectNS::EFFECT_TYPE effectType, float duration) {
 }
 
 void Entity::resetEffects() {
-	Effects* e = new Effects();
-	effects = e;
+	//Effects* e = new Effects();
+	//effects = e;
 }
+
+//void Entity::scale
```
Comment (chesterhow) [inline src/entity.cpp]: Clean up these comments

File: src/image.h [hunk 3]
```diff
@@ -63,7 +63,11 @@ class Image {
 	virtual float getY() { return spriteData.y; }
 
 	// Return scale factor.
-	virtual float getScale() { return spriteData.scale; }
+	// Return spriteData.x for convenience, because a lot of prev methods 
+	// has the same x, y scale and are already taking float as input
+	virtual float getScale() { return spriteData.scale.x; }
```
Comment (chesterhow) [inline src/image.h]: Remove this?

Comment (chesterhow) [reply src/image.h]: Or return the vector

File: src/image.h [hunk 4]
```diff
@@ -133,7 +139,7 @@ class Image {
 	// Set current frame of animation.
 	virtual void setCurrentFrame(int c);
 
-	// Set spriteData.rect to draw currentFrame
+	// Set spriteData.rect to draw currentFrascme
```
Comment (chesterhow) [inline src/image.h]: u watch out

File: src/pickupManager.cpp [hunk 5]
```diff
@@ -55,3 +60,32 @@ effectNS::EffectData PickupManager::getPickupData(effectNS::EFFECT_TYPE et) {
...
+void PickupManager::massSpawnPickups() {
+	Pickup* pickup;
+
+	for (int i = 0; i <= 10; i++) {
+		pickup = randomPickup();
+		pickup->setVelocity(randBool() ?
+		/*	VECTOR2(-pickupNS::VELOCITY, 0) :
+			VECTOR2(pickupNS::VELOCITY, 0));*/
+			VECTOR2(-300, 0) :
+			VECTOR2(300, 0));
```
Comment (chesterhow) [inline src/pickupManager.cpp]: Wat

File: src/pickupManager.h [hunk 6]
```diff
@@ -24,6 +25,20 @@ namespace effectDataNS {
...
+	//	effectNS::EffectData(effectNS::MAGNET, 0, 2.0f),
+	//	effectNS::EffectData(effectNS::INVERT, 1, 3.0f),
+	//	effectNS::EffectData(effectNS::SHIELD, 2, 0.0f),
+	//	effectNS::EffectData(effectNS::MULTIPLY, 3, 0.0f),
+	//	effectNS::EffectData(effectNS::BOOST, 4, 3.0f),
+	//	effectNS::EffectData(effectNS::SLOW, 5, 3.0f),
+	//	effectNS::EffectData(effectNS::SHRINK, 6, 3.0f),
+	//	effectNS::EffectData(effectNS::ENLARGE, 7, 3.0f),
+	//	effectNS::EffectData(effectNS::MYSTERY, 8, 0.0f)
+	//};
```
Comment (chesterhow) [inline src/pickupManager.h]: 💩

File: src/pongstar.cpp [hunk 7]
```diff
@@ -65,6 +65,10 @@ void Pongstar::initialize(HWND hwnd) {
 	Menu* menu = new Menu(audio ,input, fontManager);
 	menu->initialize(sd);
 	gameStack->push(menu);
+
+	//Classic* classic = new Classic(this, dataManager, fontManager, tmMap);
+	//classic->initialize(sd);
+	//gameStack->push(classic);
```
Comment (chesterhow) [inline src/pongstar.cpp]: 💩 💩

**Review by chesterhow — CHANGES_REQUESTED** [review]

**Comment by ivantjh** [comment]
@chesterhow

**Review by chesterhow — APPROVED** [review]

**Pull request merged**